controllers/users/authenticate-web.js

```diff
@@ -22,44 +22,67 @@
    * @type {RegisterBody}
    */
   const users = req.body;
-  users.username = 'auth_web_' + moment().format('YYYYMMDDHHmmssSSS');
+
+  const checkUser = await User.findOne({
+    where: {
+      human_id: users.human_id
+    }
+  });
 
   let insertedObject = {};
 
-  /**
-   * Inserting data to Postgres DB
-   */
-  try {
-    insertedObject = await sequelize.transaction(async (t) => {
-      const user = await UsersFunction.register(User, users, t);
-      const anonymousUser = await UsersFunction.registerAnonymous(User, user?.user_id, t);
+  if (checkUser) {
+    insertedObject = {
+      user: checkUser,
+      anonymousUser: await UsersFunction.findAnonymousUserId(User, checkUser.dataValues.user_id, {
+        raw: false
+      })
+    };
+  } else {
+    users.username = 'auth_web_' + moment().format('YYYYMMDDHHmmssSSS');
+    users.verified_status = 'UNVERIFIED';
+    /**
+     * Inserting data to Postgres DB
+     */
+    try {
+      insertedObject = await sequelize.transaction(async (t) => {
+        const user = await UsersFunction.register(User, users, t);
+        const anonymousUser = await UsersFunction.registerAnonymous(User, user?.user_id, t);
 
-      return {
-        user,
-        anonymousUser
-      };
-    });
-  } catch (e) {
-    console.log('error on sql transaction', e);
-    return res.status(500).json({
-      status: 'error on sql transaction',
-      code: 500,
-      message: e
-    });
+        return {
+          user,
+          anonymousUser
+        };
+      });
+    } catch (e) {
+      console.log('error on sql transaction', e);
+      return res.status(500).json({
+        status: 'error on sql transaction',
+        code: 500,
+        message: e
+      });
+    }
+    /**
+     * Inserting data to Postgres DB (END)
+     */
   }
-  /**
-   * Inserting data to Postgres DB (END)
-   */
 
   /**
    * Creating User to Getstream
    */
   try {
-    token = await BetterSocialCore.user.createUser(insertedObject?.user);
-    anonymousToken = Getstream.core.createToken(
-      insertedObject?.anonymousUser?.user_id?.toLowerCase()
-    );
-    await BetterSocialCore.user.createAnonymousUser(insertedObject?.anonymousUser);
+    if (checkUser) {
+      token = Getstream.core.createToken(insertedObject?.user?.user_id);
+      anonymousToken = Getstream.core.createToken(
+        insertedObject?.anonymousUser?.user_id?.toLowerCase()
+      );
+    } else {
+      token = await BetterSocialCore.user.createUser(insertedObject?.user);
+      anonymousToken = Getstream.core.createToken(
+        insertedObject?.anonymousUser?.user_id?.toLowerCase()
+      );
+      await BetterSocialCore.user.createAnonymousUser(insertedObject?.anonymousUser);
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar to the previous comment, consider using a dedicated logging service instead of `console.log`. Also, avoid sending the raw error message to the client.

```diff
-    console.log('error on inserting user to getstream creating', e);
-    return res.status(500).json({
-      status: 'error on inserting user to getstream creating',
-      code: 500,
-      message: e
-    });
+    logger.error('error on inserting user to getstream creating', e);
+    return res.status(500).json({
+      status: 'error on inserting user to getstream creating',
+      code: 500,
+      message: 'An error occurred while processing your request.'
+    });
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->